### PR TITLE
Refactor Australia holidays

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

- Refactor some Australia holidays
- Add Anzac Day to `InternationalHolidays`

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've added references to all holidays information sources used in this PR
- [x] The code style looks good: `make pre-commit` command generates no changes
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Comment by KJhellico** [comment]
@PPsyrius , thank for your suggestion! I remember seeing this holiday in some countries in Oceania. :)

File: holidays/countries/australia.py [hunk 1]
```diff
@@ -41,6 +38,11 @@ def sovereign_birthday(self) -> str:
             else "Queen's Birthday"
         )
 
+    @property
+    def boxing_day_name(self) -> str:
```
Comment (arkid15r) [inline holidays/countries/australia.py:42]: Do we really want a separate property for a single use case?

File: holidays/groups/international.py [hunk 2]
```diff
@@ -28,6 +28,21 @@ def _add_africa_day(self, name):
...
+        """
+        Add Anzac Day (April 25th)
+
+        Anzac Day is a national day of remembrance in Australia and New Zealand
+        that broadly commemorates all Australians and New Zealanders "who
+        served and died in all wars, conflicts, and peacekeeping operations"
+        and "the contribution and suffering of all those who have served".
+
+        Anzac Day is a public holiday in Australia, New Zealand, and Tonga; as
+        well as a public holiday in Samoa up until 2008.
```
Comment (arkid15r) [inline holidays/groups/international.py:41]:
```suggestion
        Anzac Day is a public holiday in Australia, New Zealand, and Tonga; it
        used to be a public holiday in Samoa up until 2008.
```

**Review by arkid15r — COMMENTED** [review]
Nice work getting rid of date/td imports!

File: holidays/countries/australia.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/australia.py:42]: Yes, it seems a bit excessive.

**Pull request merged**